### ISSUE TITLE
Model output difference between android device and desktop

### ISSUE THREAD

**Issue by sklationd** [pytorch/pytorch#48551]
## 🐛 Bug

Outputs are different between android device and desktop.
However, emulator has a same output with desktop one.
It affects to accuracy of inference in android device.

Environment:
- colab (pytorch 1.7.0)
- mobile (pytorch 1.7.0, torchvision 1.7.0)
    - emulator (Pixel XL API 30)
    - android device (Galaxy S8)  

[Pytorch discuss](https://discuss.pytorch.org/t/performance-accuracy-difference-between-desktop-and-android/104141/2)

## To Reproduce

0. I used this scripted model [quantized_best.pt](https://drive.google.com/file/d/1jNWZ3oXldM6D8-zA2ny9T7XGHy47Lew-/view?usp=sharing).
   - It is generated by below code.
``` python
from torch.utils.mobile_optimizer import optimize_for_mobile

model = torchvision.models.mobilenet_v2(pretrained=True)
model.classifier[1] = torch.nn.Linear(in_features=model.classifier[1].in_features, out_features=5)
model.eval()
model.to(device)

...
# training pipeline
...

q_model = torch.quantization.convert(model)
traced_script_module = torch.jit.trace(q_model, torch.rand(1,3,224,224).to(device))
opt_model = optimize_for_mobile(traced_script_module)
torch.jit.save(opt_model, name)
```

1. Colab
``` python
model = torch.jit.load(MYPATH, map_location='cpu')
model.eval()
one = torch.ones([1,3,224,224])
model(one)
```
``` 
// output of colab
tensor([[ 2.7785,  1.4376, -0.8041, -1.5155, -0.8308]])
```
2. Android emulator and device. Refer to [this](https://github.com/pytorch/android-demo-app/blob/f0a5bfd07f7d680b4b15c76d949abfa4bd76fa8d/PyTorchDemoApp/app/src/main/java/org/pytorch/demo/vision/ImageClassificationActivity.java#L166).
``` java
// android code snippet for testing
mModule = Module.load(MYPATH);

...

// Initialize input to one
for(int i=0;i<3*224*224;i++){
    mInputTensorBuffer.put(i,1);
}

// Get output
final Tensor outputTensor = mModule.forward(IValue.from(mInputTensor)).toTensor();
final float[] scores = outputTensor.getDataAsFloatArray();
Log.d("DEBUG", Arrays.toString(scores));
```

```
// Output of Emulator
2020-11-29 12:56:51.328 3030-3030/org.pytorch.demo D/DEBUG: [2.778521, 1.4375764, -0.80405545, -1.5155371, -0.83080786]

// Output of Actual Device
2020-11-29 12:58:08.625 725-725/org.pytorch.demo D/DEBUG: [1.224158, -0.65208256, -0.3742584, -0.123134516, -0.085230656]
```

3. Summary
``` 
// output of colab
[ 2.7785,  1.4376, -0.8041, -1.5155, -0.8308]

// output of emulator
[2.778521, 1.4375764, -0.80405545, -1.5155371, -0.83080786]

// output of actual device
[1.224158, -0.65208256, -0.3742584, -0.123134516, -0.085230656]
```

Colab has a same output with emulator, but actual device is not.
I can't figure out why this happened.

## Expected behavior

Every outputs should be same. (Colab, Emulator, Android device)

## Environment of Colab

```
Collecting environment information...
PyTorch version: 1.7.0+cu101
Is debug build: True
CUDA used to build PyTorch: 10.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.12.0

Python version: 3.6 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: 10.1.243
GPU models and configuration: GPU 0: Tesla T4
Nvidia driver version: 418.67
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.18.5
[pip3] torch==1.7.0+cu101
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.3.1
[pip3] torchvision==0.8.1+cu101
[conda] Could not collect

```

cc @jerryzh168 @jianyuh @dzhulgakov @raghuramank100 @jamesr66a @vkuzo

**Comment by supriyar** [comment]
The outputs seem quite different. Adding mobile oncall to get some help on the mobile side.

cc @kimishpatel

**Comment by vkuzo** [comment]
@sklationd , can you share the qconfig you used?  If you are running things on `fbgemm`, `reduce_range` needs to be set to True to avoid overflow in some of the kernels.

**Comment by raghuramank100** [comment]
Closing this issue, please reopen in pytorch forums as we can better support usage questions there.

**Comment by MelvinFMQ** [comment]
@raghuramank100 its not a usage question, the accuracy between the desktop and mobile inference is wildly off.

**Comment by kimishpatel** [comment]
> @sklationd , can you share the qconfig you used? If you are running things on `fbgemm`, `reduce_range` needs to be set to True to avoid overflow in some of the kernels.

@MelvinFMQ can you check this? And I assume the same app binary is run on emulator and device?
